src/secp_ops.rs

```diff
@@ -0,0 +1,73 @@
+use crate::allocator::{Allocator, NodePtr};
+use crate::cost::{check_cost, Cost};
+use crate::node::Node;
+use crate::op_utils::{atom, check_arg_count};
+use crate::reduction::{Reduction, Response};
+use k256::ecdsa::{Signature as K1Signature, VerifyingKey as K1VerifyingKey};
+use p256::ecdsa::signature::hazmat::PrehashVerifier;
+use p256::ecdsa::{Signature as P1Signature, VerifyingKey as P1VerifyingKey};
+
+const SECP256P1_VERIFY_COST: Cost = 3000000;
+const SECP256K1_VERIFY_COST: Cost = 3000000;
+
+// expects: pubkey msg sig
+pub fn op_secp256p1_verify(a: &mut Allocator, input: NodePtr, max_cost: Cost) -> Response {
+    let args = Node::new(a, input);
+    check_arg_count(&args, 3, "secp256p1_verify")?;
+
+    let cost = SECP256P1_VERIFY_COST;
+    check_cost(a, cost, max_cost)?;
+
+    // first argument is sec1 encoded pubkey
+    let pubkey = atom(args.first()?, "secp256p1_verify pubkey")?;
+    let Ok(verifier) = P1VerifyingKey::from_sec1_bytes(pubkey) else { return args.err("secp256p1_verify pubkey is not valid") };
```

Comment:
This line is really long and I can't get `cargo fmt` to fix it. I don't understand why.

This might be more rust-ic:
```suggestion
    let verifier = P1VerifyingKey::from_sec1_bytes(pubkey)
        .or_else(|_| args.err("secp256p1_verify pubkey is not valid"))?;
```

Reply:
https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html#let-else

I'd recommend using `.or_else` as it's a bit more standard and lets us use `clippy`.